### PR TITLE
nonbonded docs

### PR DESCRIPTION
**Pull request by maxentile**
Adds docstrings for `nonbonded_v3` and `convert_to_4d`.

Possible adjustments:

- [x] Remove `scales` argument?
- [ ] Add reaction-field reference?

### CODE REVIEW

File: timemachine/potentials/nonbonded.py [hunk 1]
```diff
@@ -25,6 +20,45 @@ def nonbonded_v3(
...
+            where w = cutoff * (lambda_plane_idxs + lambda_offset_idxs * lamb)
+    params : (N, 3) np.array
+        columns [charges, sigmas, epsilons], one row per particle
+    box : Optional 3x3 np.array
+    lamb : float
+    charge_rescale_mask : (N, N) np.array
+        the Coulomb contribution of pair (i,j) will be multiplied by charge_rescale_mask[i,j]
+    lj_rescale_mask : (N, N) np.array
+        the Lennard-Jones contribution of pair (i,j) will be multiplied by lj_rescale_mask[i,j]
+    scales
```
Comment (proteneer) [inline timemachine/potentials/nonbonded.py:42]: I think this can be removed.

File: timemachine/potentials/nonbonded.py [hunk 2]
```diff
@@ -25,6 +20,45 @@ def nonbonded_v3(
...
+        unused # TODO: remove?
+    beta : float
```
Comment (proteneer) [inline timemachine/potentials/nonbonded.py:44]: in OpenMM I think this is called alpha (real part PME), I'd be happy to be consistent with OpenMM

File: timemachine/potentials/nonbonded.py [hunk 3]
```diff
@@ -25,6 +20,45 @@ def nonbonded_v3(
...
+        the charge product q_ij will be multiplied by erfc(beta*d_ij)
+    cutoff : Optional float
```
Comment (proteneer) [inline timemachine/potentials/nonbonded.py:46]: at some point we should add a sanity check that U at cutoff is less than some threshold

File: timemachine/potentials/nonbonded.py [hunk 4]
```diff
@@ -25,6 +20,45 @@ def nonbonded_v3(
...
+
+    Returns
+    -------
+    energy : float
+
+    References
+    ----------
+    * Rodinger, 2005, J. Chem. Phys. "Absolute free energy calculations by thermodynamic integration in four spatial
+        dimensions" https://aip.scitation.org/doi/abs/10.1063/1.1946750
+    * TODO: Add a reference for the reaction field treatment?
```
Comment (proteneer) [inline timemachine/potentials/nonbonded.py:60]: I need to dig this up again - it's really just the real part of PME right now.

File: timemachine/potentials/nonbonded.py (hunk 1 above)
Comment (maxentile) [reply timemachine/potentials/nonbonded.py]: Addressed in https://github.com/proteneer/timemachine/pull/424/commits/eb64bf66c7ecd38516517f9e0ada9424bbffef5e (removing references) and https://github.com/proteneer/timemachine/pull/424/commits/fb5079847da403a7ab537d62a3c96cfb5bafd829 (modifying the signature)

**Pull request merged**